custom/icds_reports/utils/data_pull_scripts/ap_data_pull.sql

```diff
@@ -0,0 +1,286 @@
+
+COPY (
+select district_name, block_name, supervisor_name,awc_name,
+SUM(pse_eligible) as pse_eligible,
+SUM(CASE WHEN pse_days_attended is not null and pse_days_attended=0 then 1 ELSE 0 END) as pse_0,
+CASE WHEN SUM(pse_eligible) is not null and SUM(pse_eligible)>0 THEN SUM(CASE WHEN pse_days_attended is not null and pse_days_attended=0 then 1 ELSE 0 END)::float/SUM(pse_eligible)*100 ELSE 0 END as percent_0,
+SUM(CASE WHEN pse_days_attended is not null and pse_days_attended BETWEEN 1 and 7 then 1 ELSE 0 END) as pse_1_7,
+CASE WHEN SUM(pse_eligible) is not null and SUM(pse_eligible)>0 THEN SUM(CASE WHEN pse_days_attended is not null and pse_days_attended BETWEEN 1 and 7 then 1 ELSE 0 END)::float/SUM(pse_eligible)*100 ELSE 0 END as percent_1_7,
+SUM(CASE WHEN pse_days_attended is not null and pse_days_attended BETWEEN 8 and 14 then 1 ELSE 0 END)as  pse_8_14,
+CASE WHEN SUM(pse_eligible) is not null and SUM(pse_eligible)>0 THEN SUM(CASE WHEN pse_days_attended is not null and pse_days_attended BETWEEN 8 and 14 then 1 ELSE 0 END)::float/SUM(pse_eligible)*100 ELSE 0 END as percent_8_14,
+SUM(CASE WHEN pse_days_attended is not null and pse_days_attended between 15 and 21 then 1 ELSE 0 END)as pse_15_21,
+CASE WHEN SUM(pse_eligible) is not null and SUM(pse_eligible)>0 THEN SUM(CASE WHEN pse_days_attended is not null and pse_days_attended between 15 and 21 then 1 ELSE 0 END)::float/SUM(pse_eligible)*100 ELSE 0 END as percent_15_21,
+SUM(CASE WHEN pse_days_attended is not null and pse_days_attended>21 then 1 ELSE 0 END) as pse_gt_21,
+CASE WHEN SUM(pse_eligible) is not null and SUM(pse_eligible)>0 THEN SUM(CASE WHEN pse_days_attended is not null and pse_days_attended>21 then 1 ELSE 0 END)::float/SUM(pse_eligible)*100 ELSE 0 END as percent_gt_21,
+month
+from  awc_location inner join
+"child_health_monthly" child_health on (awc_location.doc_id=child_health.awc_id and
+    awc_location.state_id='f98e91aa003accb7b849a0f18ebd7039' and
+    district_is_test=0 and block_is_test=0 and supervisor_is_test=0 and
+    awc_is_test=0 and child_health.month='2019-10-01'
+    )
+WHERE child_health.age_in_months>36
```

Comment:
if you are going to do the join in one query it would be good to add `where aggregation_level=5` so postgres knows to skip all of the higher level rows.

Reply:
ah ok!.

Reply:
I think other levels will automatically be excluded as its an inner join, is not that true?

Reply:
they will be excluded from the final result, but they will be checked while joining. adding it to the where prevents them from even being considered